crates/ty_python_semantic/src/types/function.rs

```diff
@@ -975,7 +975,8 @@ fn is_instance_truthiness<'db>(
         | Type::TypeIs(..)
         | Type::Callable(..)
         | Type::Dynamic(..)
-        | Type::Never => {
+        | Type::Never
+        | Type::TypedDict(_) => {
```

Comment:
they're always instances of exactly `dict` at runtime, but it may not be worth the complexity trying to model that here